tools/base/BUILD

```diff
@@ -0,0 +1,16 @@
+load("@rules_python//python:pip.bzl", "compile_pip_requirements")
+load("@envoy//bazel:envoy_build_system.bzl", "envoy_package")
+
+licenses(["notice"])  # Apache 2
+
+envoy_package()
+
+compile_pip_requirements(
+    name = "requirements",
+    extra_args = [
+        "--allow-unsafe",
+        "--generate-hashes",
+        "--reuse-hashes",
+        "--resolver=backtracking",
```

Comment:
some of these might not be necessary - we have them in envoy but i think they might duplicate

Reply:
Removed `--allow-unsafe` and `--resolver=backtracking` which are both the default now according to their docs:

https://pip-tools.readthedocs.io/en/latest/#deprecations

Done in 9a7a196db825aa10991f122512b5b3661eb0a395.